### DISCUSSION TITLE
Getting ready to announce Fiboa

### DISCUSSION THREAD

**Discussion by cholmes** [fiboa/specification#16] [General]
So we managed to get a 0.1.0 release out, but it'd be good to figure out what all we want in place to start talking about Fiboa.

So far we have:
 * Fiboa core spec 0.1.0 release
 * Tillage extension 0.1.0 release
 * AI Ecosystem extension drafted (not yet released)
 * A slack channel (tge-field-boundaries) on cloud native geo slack.
 * Validation tools 
 * A couple examples of data

The absolute minimum to me would be:
 * A good blog post explaining some of the 'why' and what's next.
 * One 'real' dataset that's been converted to Fiboa - ideally with at least one or two extensions - hosted on source cooperative.
 * Posting of the video of the final session (I have it on my laptop)

And then a few things that I wouldn't block on, but it'd be great to try to get into place:

 * Some instructions / guidance on converting data & making new extensions (a clear 'call to action' for anyone interested).
 * AI Ecosystem extension released.
 * The ASU dataset converted to Fiboa + AI ecosystem extension
 * 2-3 additional datasets (could be small ones, but more 'real')
 * Finalize name, and rename slack channel to reflect it, and invite more people.
 * Some sort of tutorial that helps people learn about Fiboa in some way.
 * The pytorch plugin that Caleb and group worked on released.
 
I'm also not sure if this forum is the best place for discussions like this on the broader ecosystem. We could make like a fiboa-ecosystem repo for discussion. The advantage would be that we could then also use the issue tracker to track what's going on.

**Comment by m-mohr** [comment]
The tooling side still needs a bit of work :-)

I'm offering myself to provide a fiboa GeoParquet file converted from https://github.com/fiboa/data-survey/blob/main/data/DE-NRW.md (Germany, NRW)

For your last list, I'd love to see some work happening on the timestamps extension. This also seems curcial for the other extensions.

> We could make like a fiboa-ecosystem repo for discussion. 

We could copy the extensions repo and use the same for tooling, the tooling repos just need a different tag/topic and could be listed there.

**Reply by cholmes** [reply]
> The tooling side still needs a bit of work :-)

Ah, gotcha. Should I put something like '0.1.0 releases of tooling'? Or just something vague?

> I'm offering myself to provide a fiboa GeoParquet file converted from https://github.com/fiboa/data-survey/blob/main/data/DE-NRW.md (Germany, NRW)

Awesome! 

> For your last list, I'd love to see some work happening on the timestamps extension. This also seems curcial for the other extensions.

I'll add it to the list. Had been thinking about that, and also the 'ID' extension, and like add in at least Global FieldID. Maybe we should at least start the repos for those two and open some issues. I should probably 'clean up' the management practices extension as well - either remove it or make it clear that tillage and cover crop are their own extensions.

> We could copy the extensions repo and use the same for tooling, the tooling repos just need a different tag/topic and could be listed there.

That could be cool. I'm also just thinking about where to put 'meta' discussions like this that include tools, extensions and core together. But I do think it's less good to start with split discussions, so maybe we just start here for that. Extension repo does seem like a good place to put issues about new extensions, etc.

**Reply by m-mohr** [reply]
> Ah, gotcha. Should I put something like '0.1.0 releases of tooling'? Or just something vague?

Version numbers don't say anything. What do you want the tooling to do?
- [x] Validate GeoJSON
- [x] GeoParquet Schema
- [x] GeoParquet Data (not overly scalable for now)
- [X] Convert GeoJSON => GeoParquet
- [ ] Convert X => Y
- [ ] Viewer?
- [ ] ...?

> I'm also just thinking about where to put 'meta' discussions like this that include tools, extensions and core together.

Spec discussion forum seems fine.

> That could be cool. 

Here's the software repo + list: https://github.com/fiboa/software

> Extension repo does seem like a good place to put issues about new extensions, etc.

It's already there: https://github.com/fiboa/extensions/issues :-)

>  But I do think it's less good to start with split discussions, so maybe we just start here for that.

We just need to guide users... Make a list where to place issues.

**Reply by cholmes** [reply]
> Version numbers don't say anything. What do you want the tooling to do?

Generally work and be free of bugs? For 0.1.0 I generally just meant 'does basic validation of data'. I suppose GeoParquet data validation would be good, but I think not 'must have'. 

I'll try out the current tooling more, I don't have a great idea of what should be next. I see 'viewer' as a bit further away, but a decent command line 'info' like gpq has could be cool.

**Reply by m-mohr** [reply]
As a start to guide users into the right direction: https://github.com/fiboa/.github/blob/main/profile/README.md#where-to-start-discussions-or-raise-issues

PS: Your comment above is just a quote ;-)

**Reply by cholmes** [reply]
> We just need to guide users... Make a list where to place issues.

Cool, that sounds good. I suppose fill out the various issue templates, and then have each point at the other places to file issues / have discussions.

**Reply by m-mohr** [reply]
Issue templates and "links" to the other places in the issue trackers is also a good idea!

> Generally work and be free of bugs?

Haha, free of bugs :-) I'll try...

> but a decent command line 'info' like gpq has could be cool.

Good idea.
